File: CriminalIntent/app/src/main/java/kg/dostek/criminalintent/CrimePagerActivity.java
package kg.dostek.criminalintent;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
//import android.support.annotation.Nullable;
//import android.support.v4.app.Fragment;
//import android.support.v4.app.FragmentActivity;
//import android.support.v4.app.FragmentManager;
//import android.support.v4.app.FragmentStatePagerAdapter;
//import android.support.v4.view.ViewPager;

import java.util.List;
import java.util.UUID;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;
import androidx.viewpager.widget.ViewPager;

import static androidx.fragment.app.FragmentStatePagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT;

/**
 * Created by Alexander on 09.01.2017.
 */

public class CrimePagerActivity extends FragmentActivity {
    public static final String EXTRA_CRIME_ID = "kg.dostek.criminalintent.crime_id";

    private ViewPager mViewPager;
    private List<Crime> mCrimes;

    public static Intent newIntent(Context packageContext, UUID crimeId) {
        Intent intent = new Intent(packageContext, CrimePagerActivity.class);
        intent.putExtra(EXTRA_CRIME_ID, crimeId);
        return intent;
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_crime_pager);

        UUID crimeId = (UUID) getIntent().getSerializableExtra(EXTRA_CRIME_ID);

        mViewPager = (ViewPager) findViewById(R.id.crime_view_pager);
        mCrimes = CrimeLab.get(this).getCrimes();

        //mViewPager.setOffscreenPageLimit();
        FragmentManager fragmentManager = getSupportFragmentManager();
        mViewPager.setAdapter(new FragmentStatePagerAdapter(fragmentManager, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
            @Override
            public Fragment getItem(int position) {
                Crime crime = mCrimes.get(position);
                return CrimeFragment.newInstance(crime.getId());
            }

            @Override
            public int getCount() {
                return mCrimes.size();
            }
        });

        for (int i = 0; i < mCrimes.size(); i++)
            if (mCrimes.get(i).getId().equals(crimeId))
                mViewPager.setCurrentItem(i);
    }
}

File: CriminalIntent/app/src/main/kotlin/kt/akimov/criminalintent/presentation/crimepager/CrimePagerActivity.kt
package kt.akimov.criminalintent.presentation.crimepager

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.viewpager.widget.ViewPager
import androidx.viewpager2.adapter.FragmentStateAdapter
import kg.dostek.criminalintent.R
import kt.akimov.criminalintent.data.CrimeLab
import kt.akimov.criminalintent.domain.models.CrimeItem
import kt.akimov.criminalintent.presentation.crimeitem.CrimeFragment
import java.util.*


class CrimePagerActivity : AppCompatActivity() {

    //lateinit var viewPager: ViewPager2
    lateinit var viewPager: ViewPager
    //private lateinit var pagerAdapter: FragmentStateAdapter
    private lateinit var viewModel: CrimePagerViewModel
    private lateinit var crimes: List<CrimeItem>

    companion object {
        private const val EXTRA_CRIME_ID = "kt.akimov.criminalintent.crime_id"
        fun newIntent(packageContext: Context?, crimeId: UUID): Intent {
            val intent = Intent(packageContext, CrimePagerActivity::class.java)
            intent.putExtra(EXTRA_CRIME_ID, crimeId)
            return intent
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_crime_pager)

        val crimeId = intent?.getSerializableExtra(EXTRA_CRIME_ID) as UUID

        viewPager = findViewById(R.id.crime_view_pager)
        crimes = CrimeLab.getCrimes()
        //viewPager.setPageTransformer(ZoomOutPageTransformer())
        //mViewPager.setOffscreenPageLimit();

        val fragmentManager = supportFragmentManager
        // var localAdapter = CrimePagerAdapter(this)
        var localAdapter = object : FragmentStatePagerAdapter(fragmentManager, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

            override fun getItem(position: Int): Fragment {
                var crime = crimes[position]

                return CrimeFragment.newInstance(crime.id)
            }

            override fun getCount(): Int {
                return crimes.size
            }
        }
        viewPager.adapter = localAdapter
        //viewPager.addOnPageChangeListener(ViewPager.OnPageChangeListener{})

        viewModel = ViewModelProviders.of(this).get(CrimePagerViewModel::class.java)

        // setupViewPager()
        // subscribeObservers()
        // loadData()

        for (i in crimes.indices) {
            if (crimes[i].id == crimeId) {
                viewPager.currentItem = i
                break
            }
        }
    }

    private fun loadData() {
        viewModel.getCrimes()
    }

    private fun subscribeObservers() {
        val crimeObserver = Observer<List<CrimeItem>> {
            //  pagerAdapter.setItems(it)
        }
        viewModel.observableCrimeItems.observe(this, crimeObserver)
    }

    private fun setupViewPager() {

        //CrimeFragmentPagerAdapter(this)

        var localAdapter = CrimePagerAdapter(this)


        /*pagerAdapter = object : FragmentStateAdapter(this) {

            override fun getItemCount(): Int {
                return crimes.size
            }

            override fun createFragment(position: Int): Fragment {
                var crime = crimes[position]

                return CrimeFragment.newInstance(crime.id)
            }
        }*/

        // viewPager.adapter = localAdapter
    }

    private inner class CrimePagerAdapter(fa: AppCompatActivity) : FragmentStateAdapter(this) {

        override fun getItemCount(): Int {
            return crimes.size
        }

        override fun createFragment(position: Int): Fragment {
            var crime = crimes[position]

            return CrimeFragment.newInstance(crime.id)
        }
    }
}